### PR TITLE
Adding support for TAN67 PCB

### PR DESCRIPTION
**Pull request by Pangorin**
## Description

Adding support for TAN67 PCB, a 65% blocker PCB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
There was no need to open a new PR, all that has done is fragment review efforts (and bump it to the bottom of the list).

**Comment by Pangorin** [comment]
@zvecr I based this new pull request on the old pull request (with your reviews and other collaborators reviews). I just want to make sure that I didn't mess things up this time, and I checked all files carefully :D

**Comment by zvecr** [comment]
Thats not the point. All of that context is now elsewhere, which makes it harder for reviewers.

It also looks like you were asked not to create a new PR on discord too. You were asked to be patient with the review process.

File: keyboards/pangorin/tan67/config.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#ifdef RGB_MATRIX_ENABLE
+    /* RGB Matrix config */
+    #define RGB_DI_PIN B15
```
Comment (drashna) [inline keyboards/pangorin/tan67/config.h:20]:
```suggestion
    #define RGB_DI_PIN B15
    #define WS2812_DI_PIN B15
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by Pangorin — COMMENTED** [review]
Update config.h file.

File: keyboards/pangorin/tan67/readme.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# TAN67 PCB
+
+A 65% PCB, which controlled by an STM32F072 chipset.
+
+* Keyboard Maintainer: [Pangorin](https://github.com/pangorin)
+* Hardware Supported: STM32F072
```
[the remark this answers is not in the document]
Comment (Pangorin) [reply keyboards/pangorin/tan67/readme.md:6]: Yes. The maker of this PCB (the keyboard named TAN67) didn't make this keyboard anymore, so no PCB to buy.

File: keyboards/pangorin/tan67/config.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
```
Comment (drashna) [inline keyboards/pangorin/tan67/config.h:20]:
```suggestion
```

**Review by drashna — APPROVED** [review]

File: keyboards/pangorin/tan67/info.json [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    "bootloader": "stm32-dfu",
+    "features": {
+        "bootmagic": true,
+        "mousekey": false,
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": true,
+        "rgb_matrix": true
+    },
```
Comment (fauxpark) [inline keyboards/pangorin/tan67/info.json:25]:
```suggestion
    },
    "ws2812": {
        "pin": "B15"
    },
```

File: keyboards/pangorin/tan67/config.h [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#ifdef RGB_MATRIX_ENABLE
+    /* RGB Matrix config */
+    #define WS2812_DI_PIN B15
```
Comment (fauxpark) [inline keyboards/pangorin/tan67/config.h:20]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/pangorin/tan67/info.json [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+            {"x": 19, "y": 8, "flags": 2},
+            {"x": 52, "y": 8, "flags": 2},
+            {"x": 90, "y": 8, "flags": 2},
+            {"x": 127, "y": 8, "flags": 2},
+            {"x": 164, "y": 8, "flags": 2},
+            {"x": 205, "y": 8, "flags": 2}
+        ]
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (dunk2k) [inline keyboards/pangorin/tan67/info.json:168]:
```suggestion
    },
    "community_layouts": ["65_ansi_blocker"],
    "layouts": {
        "LAYOUT_65_ansi_blocker": {
```

File: keyboards/pangorin/tan67/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+└───────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└───────────┘└────┘
+┌───────────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌───────┐┌────┐┌────┐
+│           ││    ││    ││    ││    ││    ││    ││    ││    ││    ││    ││       ││Vai+││    │
+└───────────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└───────┘└────┘└────┘
+┌─────┐┌─────┐┌─────┐┌─────────────────────────────────────┐┌─────┐┌─────┐  ┌────┐┌────┐┌────┐
+│     ││     ││     ││                                     ││     ││     │  │    ││Vai-││    │
+└─────┘└─────┘└─────┘└─────────────────────────────────────┘└─────┘└─────┘  └────┘└────┘└────┘
+*/
+    /*  Row:    0        1        2        3        4        5        6        7        8        9        10       11       12       13       14     */
+    [_FN1]  = LAYOUT(
```
Comment (dunk2k) [inline keyboards/pangorin/tan67/keymaps/default/keymap.c:73]:
```suggestion
    [_BASE] = LAYOUT_65_ansi_blocker(
                KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,     KC_0,    KC_MINS,  KC_EQL,  KC_BSPC, KC_HOME,
                KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,     KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS, KC_PGUP,
                KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,     KC_SCLN, KC_QUOT,  KC_ENT,           KC_PGDN,
                KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,   KC_SLSH, KC_RSFT,  KC_UP,            KC_DEL,
                KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,                             MO(_FN1), KC_RCTL, KC_LEFT,  KC_DOWN, KC_RGHT
            ),
/*
┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────────┐┌────┐
│`   ││F1  ││F2  ││F3  ││F4  ││F5  ││F6  ││F7  ││F8  ││F9  ││F10 ││F11 ││F12 ││Reset   ││    │
└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────────┘└────┘
┌──────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌──────┐┌────┐
│      ││    ││Togg││    ││    ││    ││    ││    ││    ││    ││    ││    ││    │|      ││    │
└──────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└──────┘└────┘
┌───────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌───────────┐┌────┐
│       ││    ││    ││    ││    ││    ││    ││    ││    ││    ││    ││    ││           ││    │
└───────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└───────────┘└────┘
┌───────────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌───────┐┌────┐┌────┐
│           ││    ││    ││    ││    ││    ││    ││    ││    ││    ││    ││       ││Vai+││    │
└───────────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└───────┘└────┘└────┘
┌─────┐┌─────┐┌─────┐┌─────────────────────────────────────┐┌─────┐┌─────┐  ┌────┐┌────┐┌────┐
│     ││     ││     ││                                     ││     ││     │  │    ││Vai-││    │
└─────┘└─────┘└─────┘└─────────────────────────────────────┘└─────┘└─────┘  └────┘└────┘└────┘
*/
    /*  Row:    0        1        2        3        4        5        6        7        8        9        10       11       12       13       14     */
    [_FN1]  = LAYOUT_65_ansi_blocker(
```

File: keyboards/pangorin/tan67/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+└───────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└───────────┘└────┘
+┌───────────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌───────┐┌────┐┌────┐
+│           ││    ││    ││    ││    ││    ││    ││    ││    ││    ││    ││       ││Vai+││    │
+└───────────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└───────┘└────┘└────┘
+┌─────┐┌─────┐┌─────┐┌─────────────────────────────────────┐┌─────┐┌─────┐  ┌────┐┌────┐┌────┐
+│     ││     ││     ││                                     ││     ││     │  │    ││Vai-││    │
+└─────┘└─────┘└─────┘└─────────────────────────────────────┘└─────┘└─────┘  └────┘└────┘└────┘
+*/
+    /*  Row:    0        1        2        3        4        5        6        7        8        9        10       11       12       13       14     */
+    [_FN1]  = LAYOUT(
```
Comment (dunk2k) [inline keyboards/pangorin/tan67/keymaps/via/keymap.c:73]:
```suggestion
    [_BASE] = LAYOUT_65_ansi_blocker(
                KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,     KC_0,    KC_MINS,  KC_EQL,  KC_BSPC, KC_HOME,
                KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,     KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS, KC_PGUP,
                KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,     KC_SCLN, KC_QUOT,  KC_ENT,           KC_PGDN,
                KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,   KC_SLSH, KC_RSFT,  KC_UP,            KC_DEL,
                KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,                             MO(_FN1), KC_RCTL, KC_LEFT,  KC_DOWN, KC_RGHT
            ),
/*
┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────────┐┌────┐
│`   ││F1  ││F2  ││F3  ││F4  ││F5  ││F6  ││F7  ││F8  ││F9  ││F10 ││F11 ││F12 ││Reset   ││    │
└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────────┘└────┘
┌──────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌──────┐┌────┐
│      ││    ││Togg││    ││    ││    ││    ││    ││    ││    ││    ││    ││    │|      ││    │
└──────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└──────┘└────┘
┌───────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌───────────┐┌────┐
│       ││    ││    ││    ││    ││    ││    ││    ││    ││    ││    ││    ││           ││    │
└───────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└───────────┘└────┘
┌───────────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌────┐┌───────┐┌────┐┌────┐
│           ││    ││    ││    ││    ││    ││    ││    ││    ││    ││    ││       ││Vai+││    │
└───────────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└────┘└───────┘└────┘└────┘
┌─────┐┌─────┐┌─────┐┌─────────────────────────────────────┐┌─────┐┌─────┐  ┌────┐┌────┐┌────┐
│     ││     ││     ││                                     ││     ││     │  │    ││Vai-││    │
└─────┘└─────┘└─────┘└─────────────────────────────────────┘└─────┘└─────┘  └────┘└────┘└────┘
*/
    /*  Row:    0        1        2        3        4        5        6        7        8        9        10       11       12       13       14     */
    [_FN1]  = LAYOUT_65_ansi_blocker(
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by Pangorin** [comment]
Fixing...

File: keyboards/pangorin/tan67/info.json [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+        "console": false,
+        "command": false,
+        "nkro": true,
+        "rgb_matrix": true
+    },
+    "ws2812": {
+        "pin": "B15"
+    },
+    "rgb_matrix": {
+        "driver": "WS2812",
```
Comment (dunk2k) [inline keyboards/pangorin/tan67/info.json:30]:
```suggestion
        "driver": "WS2812",
        "max_brightness": 200,
```

File: keyboards/pangorin/tan67/config.h [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#ifdef RGB_MATRIX_ENABLE
+    /* RGB Matrix config */
+    #define RGB_MATRIX_LED_COUNT 84
+    #define RGB_MATRIX_MAXIMUM_BRIGHTNESS 200
```
Comment (dunk2k) [inline keyboards/pangorin/tan67/config.h:21]: migrated to `info.json`
```suggestion
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Comment by nagiaoe2004** [comment]
i have the pcb Tan67 but not using STM32 it uisng Atmega32u4.,i dont have see file Tan67 Amega32u4,i need for flash QMK
and using Via

**Comment by Pangorin** [comment]
> i have the pcb Tan67 but not using STM32 it uisng Atmega32u4.,i dont have see file Tan67 Amega32u4,i need for flash QMK
> and using Via

I assume that your PCB is Rev.1 (or R1, mine is Rev.2). And I just have the Rev.2 so unfortunately, no support for Rev.1 PCB (the one using ATMega32u4)

**Comment by nagiaoe2004** [comment]
So is there any way to rebuild for ATmega 32u4?

**Comment by keyboard-magpie** [comment]
Not using this code, no.

**Comment by nagiaoe2004** [comment]
> Not using this code, no.

i mean code another for atmega32u4  to flash

**Comment by nagiaoe2004** [comment]
> > i have the pcb Tan67 but not using STM32 it uisng Atmega32u4.,i dont have see file Tan67 Amega32u4,i need for flash QMK
> > and using Via
> 
> I assume that your PCB is Rev.1 (or R1, mine is Rev.2). And I just have the Rev.2 so unfortunately, no support for Rev.1 PCB (the one using ATMega32u4)

it ATMega32U4 it need change all file or some file?

**Comment by Pangorin** [comment]
> > > i have the pcb Tan67 but not using STM32 it uisng Atmega32u4.,i dont have see file Tan67 Amega32u4,i need for flash QMK
> > > and using Via
> > 
> > 
> > I assume that your PCB is Rev.1 (or R1, mine is Rev.2). And I just have the Rev.2 so unfortunately, no support for Rev.1 PCB (the one using ATMega32u4)
> 
> it ATMega32U4 it need change all file or some file?

Yes, I believe that the Rev.1 is totally different compared to Rev.2 (matrix, and of course the MCU itself).

**Pull request merged**